day_7/high_level.md

```diff
@@ -0,0 +1,23 @@
+
+Ok at this point we have all the string, broken down into individual characters, index position found, X number of index positions added (or subtracted), found the index position of the new integer, and converted back to the value associated with that index position.  To return it back to one beautiful non-sensical string the `.join` method will have to be used **AFTER** all that has been completed and we should get an output of an encrypted message.
+
+
+
+## Checker Board
+This one doesn't jump out to me quite as quickly on what to do as the cipher.  Firstly we will need to prompt the user for input on number of rows/columns.  Will want to check to make sure the info entered was an even number and if not re-prompt for an even number.  Then run that even-numbered integer into a function where it prints a row alternating between `'X'` and `' '`.  It will then have to print x-1 amount of rows switching off with the starting position and alternating from there.
+
+I'm not sure if I'll want to use while loops or just use the .each method or both.  This is one that I will have to do to really understand what I need to do fully.  *Looks like it will definitely be a loop.*  
+
+First thing I did was create the method to ask the user for input.  If the user inputs an odd number the method reruns itself until an even number is inputed.  If a string is entered the program ends. I saved the input from that method in a global variable then set two global variables to equal "X " and " X" to make my rows and columns filler.  
+
+Then I created another method using the saved user input as a parameter to print out the board.  The inputed number is divided by 2 at three points in the last method because my method defaults to printing two rows every iteration with two columns also filled each iteration.  I feel this if fine because checkerboards are always an even number of rows/columns.  I probably could reconfigure some things to make odd numbers of rows/columns acceptable.  Mainly saving the variables as `"x"`  and `" "` and then reconfiguring how it reads the block in the while loop and the while loop itself.  For now I'm going to leave it for the above reason.
+
+####Update
+I went back and changed the code to allow for odd numbers to be accepted.  Personally this goes against my beliefs on how a checkerboard is defined but life is about growth.
```

Comment:
Excellent mindset! I know you wrote that kind of in jest, but it is true that we are learning to bend our programs to our will, so the extent to which they replicate reality is up to us!